### PR TITLE
fix(config-ui): change jira board selection box to be searchable

### LINKED ISSUE #3254: [Question][Blueprints] Can't get all Jira Boards
## Question
When setting the jira data scope，I can only get 50 jira boards to choose, actually we have more than 50 boards.How can I get the others?

### PR DESCRIPTION
**Pull request by mintsweet**
# Summary

Change the Jira board selection box to be searchable, and avoid the interface only returning 50 boards limit.

### Does this close any open issues?
Closes #3254

### Screenshots
screenshot-20220930-120743

### Other Information
Nothing.

### CODE REVIEW

File: config-ui/src/hooks/useDataScopesManager.jsx [hunk 1]
```diff
@@ -458,14 +458,15 @@ function useDataScopesManager({
     (boardIds = [], boardListItems = []) => {
       return boardIds.map((bId, sIdx) => {
         const boardObject = boardListItems.find(
-          (apiBoard) => Number(apiBoard.id) === Number(bId)
+          (apiBoard) =>
+            Number(apiBoard.id) === Number(bId) || +apiBoard.boardId === +bId
```
Comment (likyh) [inline config-ui/src/hooks/useDataScopesManager.jsx:462]: what difference between these 2 conditions?

File: config-ui/src/pages/blueprints/blueprint-settings.jsx [hunk 2]
```diff
@@ -672,7 +674,10 @@ const BlueprintSettings = (props) => {
                 [Providers.JIRA].includes(c.plugin)
                   ? getJiraMappedBoards(
                       c.scope.map((s) => s.options?.boardId),
-                      allJiraResources?.boards
+                      [
+                        ...allJiraResources?.boards,
+                        ...c.scope.map((s) => s.options)
```
Comment (likyh) [inline config-ui/src/pages/blueprints/blueprint-settings.jsx:679]: this change for what?

File: config-ui/src/pages/blueprints/blueprint-settings.jsx [hunk 3]
```diff
@@ -897,35 +905,64 @@ const BlueprintSettings = (props) => {
...
-                  boardsList: jiraResourcesResponse?.boards
-                    ? getJiraMappedBoards(
-                        c.boardIds,
-                        jiraResourcesResponse?.boards
-                      )
-                    : []
-                })
-            )
+      fetchIssueTypes()
+      fetchFields()
```
Comment (likyh) [inline config-ui/src/pages/blueprints/blueprint-settings.jsx:909]: why fetch these two when fetch boards?

**Review by likyh — COMMENTED** [review]
some other bugs:
1. when open modal agian, the board name lose
2. (main also exist) When I delete a board or adjust the order of the boards, the transformation of the second page will not match. For example, the original requirement of board1 is tagA, and the requirement of board2 is tagB. After 1 is deleted, board2 becomes tagA

**Comment by likyh** [comment]
Maybe we can change this tip "Select the boards you would like to sync."

**Comment by likyh** [comment]
I find a bug again:

3. type mapping is wrong when edit board
image

**Review by likyh — APPROVED** [review]
I test it and cannot find new bugs compared with the main branch. So Approved.

File: config-ui/src/components/blueprints/DataScopesGrid.jsx [hunk 4]
```diff
@@ -171,8 +171,8 @@ const DataScopesGrid = (props) => {
                     padding: 0
                   }}
                 >
-                  {c.boards.map((board, bIdx) => (
```
Comment (e2corporation) [inline config-ui/src/components/blueprints/DataScopesGrid.jsx:174]: @mintsweet  @likyh This is a bug, this should not have been changed. `boards` Represents the Boards Associated with the Blueprints. `boardsList` represents ALL Boards

Comment (likyh) [reply config-ui/src/components/blueprints/DataScopesGrid.jsx:174]: It looks like boards is string array and boardsList is object list. not boardsList represents ALL Boards but boards represents a part of.

**Comment by e2corporation** [comment]
> some other bugs:
> 
> 1. when open modal agian, the board name lose
@likyh This was an advanced timing issue that's existed for sometime. JIRA API data is not ready on the first modification request so the active board names are not applied, on the second modification request the proxy data is ready so true board name are displayed. Our Design requirements have only ever identified Boards using "Board [ID]", our blueprint only stores board ID as well. So this is why I've left this behavior intact until the timing issue is resolved. The Important thing is that Board ID is still preserved, regardless if actual name or generic board name is displayed.

> 2. (main also exist) When I delete a board or adjust the order of the boards, the transformation of the second page will not match. For example, the original requirement of board1 is tagA, and the requirement of board2 is tagB. After 1 is deleted, board2 becomes tagA
This was recently fixed by my last PR

File: config-ui/src/components/blueprints/DataScopesGrid.jsx (hunk 4 above)
Comment (e2corporation) [reply config-ui/src/components/blueprints/DataScopesGrid.jsx:174]: `boards` is the retrieved list of stored boards on the blueprint, designed for Display on the Data Scopes Grid.

**Comment by likyh** [comment]
> > some other bugs:
> > 
> > 1. when open modal agian, the board name lose
> 
> @likyh This was an advanced timing issue that's existed for sometime. JIRA API data is not ready on the first modification request so the active board names are not applied, on the second modification request the proxy data is ready so true board name are displayed. Our Design requirements have only ever identified Boards using "Board [ID]", our blueprint only stores board ID as well. So this is why I've left this behavior intact until the timing issue is resolved. The Important thing is that Board ID is still preserved, regardless if actual name or generic board name is displayed.
> 
> > 2. (main also exist) When I delete a board or adjust the order of the boards, the transformation of the second page will not match. For example, the original requirement of board1 is tagA, and the requirement of board2 is tagB. After 1 is deleted, board2 becomes tagA
> 
> This was recently fixed by my last PR for "Blueprint Settings Degraded..."

1 is after the bug fix you said and it's a new bug on this PR and it fixed.

2 is https://github.com/apache/incubator-devlake/issues/3294

File: config-ui/src/components/blueprints/DataScopesGrid.jsx (hunk 4 above)
Comment (e2corporation) [reply config-ui/src/components/blueprints/DataScopesGrid.jsx:174]: This needs to be reverted to `c.boards.map((board, bIdx)`

**Comment by e2corporation** [comment]
Where does the board limit of 50 come from in the first place?

**Pull request merged**